### PR TITLE
DOC add docs for cos7

### PR DESCRIPTION
**Pull request by beckermr**
This PR adds docs and an announcement for cos7 and the new compilers.

### CODE REVIEW

**Comment by beckermr** [comment]
cc @isuruf

**Comment by beckermr** [comment]
@conda-forge/core any comments here or shall we merge?

File: src/maintainer/knowledge_base.rst [hunk 1]
```diff
@@ -1007,3 +1009,19 @@ To skip the pypy builds, do the following,
...
+
+To use the newer CentOS 7 ``sysroot`` with ``glibc`` ``2.17`` on ``linux-64``,
+put the following in your build section
+
+.. code-block:: yaml
+
+   requirements:
+     build:
+       - {{ compiler('c') }}
+       - sysroot_linux-64 2.17  # [linux]
```
Comment (isuruf) [inline src/maintainer/knowledge_base.rst:1025]:
```suggestion
       - sysroot_linux-64 2.17  # [linux64]
```

**Comment by beckermr** [comment]
I don't actually know how to do that myself (and I didn't realize we had already built one - I assumed I would have to do it later). I'll do my best but any help on adding that would be much appreciated.

**Comment by beckermr** [comment]
ok @isuruf @conda-forge/core this is ready for another look!

File: src/maintainer/knowledge_base.rst [hunk 2]
```diff
@@ -1007,3 +1009,30 @@ To skip the pypy builds, do the following,
...
+
+.. code-block:: yaml
+
+   requirements:
+     build:
+       - {{ compiler('c') }}
+       - sysroot_linux-64 2.17  # [linux64]
+
+You also need to use a newer docker image by setting this in the ``conda_build_config.yaml``
+of your recipe
```
Comment (isuruf) [inline src/maintainer/knowledge_base.rst:1030]:
```suggestion
of your recipe and rerendering.
```

File: src/maintainer/knowledge_base.rst [hunk 3]
```diff
@@ -1007,3 +1009,30 @@ To skip the pypy builds, do the following,
...
      skip: True         # [python_impl == 'pypy']
+
+
+.. _centos7:
+
+Using CentOS 7
+==============
+
+To use the newer CentOS 7 ``sysroot`` with ``glibc`` ``2.17`` on ``linux-64``,
+put the following in your build section
```
Comment (isuruf) [inline src/maintainer/knowledge_base.rst:1020]:
```suggestion
put the following in your build section.
```

**Pull request merged**